lib/elixir/pages/anti-patterns/code-anti-patterns.md

```diff
@@ -47,7 +47,38 @@ Elixir makes a clear distinction between **documentation** and code comments. Th
 
 ## Long parameter list
 
-TODO.
+#### Problem
+
+In a functional language like Elixir, functions tend to be pure and therefore do not access or manipulate values outside their scopes. Since the input values of these functions are just their parameters, it is natural to expect that functions with a long list of parameters be created to keep them pure. However, when this list has **more than three or four parameters**, the function's interface becomes confusing and prone to errors during use.
```

Comment:
> functions tend to be pure

I wouldn't focus on purity, since this is not so true in Elixir. Immutability is a subset of purity, and the only one relevant for this explanation I think?

Actually, this is almost true for any language disregarding the paradigm I think, so maybe we should just focus on what's the reason ("interface becomes confusing and prone to errors during use.") rather than what it isn't (fear of mutations).

Reply:
Agreed, the problem here is IMO absolutely not related to the fact that Elixir is functional. A function (or method) with too many arguments is a smell in any language in my opinion. I think the point of this entry in the code smells is to explain readers how to address this smell in Elixir.

Reply:
I agree that this smell is independent of language or paradigm, so much so that in the original catalog proposed in my research it is found in the "Traditional" category, which are the most generic smells.  

Anyway, that comment about the purity of functions in the introduction to the explanation just refers to a tendency of functional languages like Elixir to have functions with more parameters than in OO.  Let it be clear that this is not a problem, after all every smell is context sensitive, which means that not every function that has many parameters will represent a quality problem... this is just a warning sign that should be evaluated by the developer.  :)